backend/src/routes/users.js

```diff
@@ -1,15 +1,92 @@
+const auth = require('../auth');
 const express = require('express');
 const router = express.Router();
 const models = require('../models');
+const db = require('../db');
+const utils = require('../utils');
+const randtoken = require('rand-token');
 
-router.get('/:id', (req, res) => {
-    res.send('returning specific user ' + req.params.id);
-});
+const getUser = async (req, res) => {
+    try {
+        const phoneNumber = req.body.phoneNumber;
+        const user = await db.users.getUser(phoneNumber);
+        res.status(200).send(user);
+    } catch (err) {
+        res.status(420).send('Failed to get user: ' + err);
+    }
+};
 
-router.get('/', (req, res) => {
-    models.users.findAll({}).then(function(users) {
+const getAllUsers = async (req, res) => {
+    try {
+        const users = await db.users.getAllUsers();
         res.status(200).send(users);
-    });
-});
+    } catch (err) {
+        res.status(420).send('Failed to get user list: ' + err);
+    }
+};
+
+const updateUser = async (req, res) => {
+    try {
+        const user = req.body;
+        await db.users.updateUser(user);
+        res.status(200).send('Updated user details');
+    } catch (err) {
+        res.status(420).send('Failed to update user details: ' + err);
+    }
+};
+
+const deleteUser = async (req, res) => {
+    try {
+        const phoneNumber = req.body.phoneNumber;
```

Comment:
This allows me to remove any user in the application, right? I think you should only be able to remove your own user so it should not be based on the phone number provided as parameter but instead `req.user` AKA the authenticated user.